File: src/api/feed.ts
import { callCommand } from "./tauri";
import type { FeedItem } from "./types";
import type { UpdateFeedItemStateInput } from "./generated/UpdateFeedItemStateInput";

// Input types GENERATED via ts-rs (ADR 0048).
export type { UpdateFeedItemStateInput } from "./generated/UpdateFeedItemStateInput";

export function listFeedItems() {
  return callCommand<FeedItem[]>("list_feed_items");
}

export function updateFeedItemState(input: UpdateFeedItemStateInput) {
  return callCommand<FeedItem>("update_feed_item_state", { input });
}

File: src-tauri/src/commands/feed.rs
use crate::{app_state, storage};

#[tauri::command]
pub fn list_feed_items(
    state: tauri::State<'_, app_state::AppState>,
) -> Result<Vec<storage::FeedItem>, String> {
    state.list_feed_items().map_err(|error| error.to_string())
}

#[tauri::command]
pub fn update_feed_item_state(
    input: storage::FeedItemStateInput,
    state: tauri::State<'_, app_state::AppState>,
) -> Result<storage::FeedItem, String> {
    state
        .update_feed_item_state(input)
        .map_err(|error| error.to_string())
}
